crates/sui-verifier/src/entry_points_verifier.rs

```diff
@@ -140,12 +140,15 @@ fn verify_init_function(module: &CompiledModule, fdef: &FunctionDefinition) -> R
         Ok(())
     } else {
         Err(format!(
-            "Expected last (and at most second) parameter for {}::{} to be &mut {}::{}::{}, but found {}",
+            "Expected last (and at most second) parameter for {}::{} to be &mut {}::{}::{} or &{}::{}::{}, but found {}",
             module.self_id(),
             INIT_FN_NAME,
             SUI_FRAMEWORK_ADDRESS,
             TX_CONTEXT_MODULE_NAME,
             TX_CONTEXT_STRUCT_NAME,
+            SUI_FRAMEWORK_ADDRESS,
+            TX_CONTEXT_MODULE_NAME,
+            TX_CONTEXT_STRUCT_NAME,
             format_signature_token(view, &parameters[0]),
         ))
```

Comment:
Consider re-using the earlier parameters?

```suggestion
        Err(format!(
            "Expected last (and at most second) parameter for {0}::{1} to be &mut {2}::{3}::{4} or &{2}::{3}::{4}, but found {5}",
            module.self_id(),
            INIT_FN_NAME,
            SUI_FRAMEWORK_ADDRESS,
            TX_CONTEXT_MODULE_NAME,
            TX_CONTEXT_STRUCT_NAME,
            format_signature_token(view, &parameters[0]),
        ))
```